File: 백준/Silver/11723. 집합/집합.py
import sys

input = sys.stdin.readline
m = int(input())
s = set()

for _ in range(m):
  operation = input().strip().split(" ")
  op = operation[0]
  if len(operation) == 2:
    n = int(operation[1])
    if op == "add":
      s.add(n)
    elif op == "remove":
      s.discard(n)
    elif op == "check":
      print(1 if n in s else 0)
    elif op == "toggle":
      if n in s:
        s.discard(n)
      else:
        s.add(n)
  else:
    if op == "all":
      s = set(range(1, 21))
    elif op == "empty":
      s = set()

File: 백준/Silver/11723. 집합/집합.java
import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		int m = Integer.parseInt(br.readLine());
		int s = 0;
		
		for(int i = 0; i < m; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			String c = st.nextToken();
			
			switch (c) {
				case "add": {
					int x = Integer.parseInt(st.nextToken());
					s = s | (1 << x);
					break;
				}
				case "remove": {
					int x = Integer.parseInt(st.nextToken());
					s = s & ~(1 << x);
					break;
				}
				case "check": {
					int x = Integer.parseInt(st.nextToken());
					
					sb.append((s & (1 << x)) != 0 ? "1\n" : "0\n");
					break;
				}			
				case "toggle": {
					int x = Integer.parseInt(st.nextToken());
					s = s ^ (1 << x);
					break;
				}	
				case "all": {
					s = (1 << 21) - 1;
					break;
				}
				case "empty": {
					s = 0;
					break;
				}
			}
		}
		System.out.println(sb);
	}
}